src/main/api/publication.ts

```diff
@@ -122,8 +126,14 @@ export class PublicationApi {
         const newDocs = [];
 
         for (const path of paths) {
-            const newDoc = await this.catalogService.importFile(path);
-            newDocs.push(newDoc);
+            try {
+                const newDoc = await this.catalogService.importFile(path);
+                newDocs.push(newDoc);
```

Comment:
The signature of the `importFile()` function indicates `Promise<PublicationDocument>` return type, but the function body can in fact return `null` (when `strictNullChecks` is enabled in `tsconfig.json`, the TypeScript compiler will detect this coding error).

So, I would suggest not pushing `newDoc` into the array because otherwise this edge-case will crash `convertDocumentToView()`.